src/tools/generators.ts

```diff
@@ -245,3 +259,279 @@ export function installGenerators(generators: string[]): string[] {
+    { platform: Platforms.Expo, type: "mobile", name: "app-icon-ios.png", inputFile: "ios-universal.png" },
+    { platform: Platforms.Expo, type: "mobile", name: "app-icon-android-legacy.png", inputFile: "android-legacy.png" },
+    { platform: Platforms.Expo, type: "mobile", name: "app-icon-android-adaptive-background.png", inputFile: "android-adaptive-background.png" },
+    { platform: Platforms.Expo, type: "mobile", name: "app-icon-android-adaptive-foreground.png", inputFile: "android-adaptive-foreground.png" },
+    { platform: Platforms.Expo, type: "mobile", name: "app-icon-all.png", inputFile: "ios-universal.png" },
+    { platform: Platforms.Expo, type: "web", name: "app-icon-web-favicon.png", inputFile: "ios-universal.png" },
+  ],
+  rules: [
+    { platform: Platforms.Ios, size: { universal: 1024 }, scale: 1, idiom: "ios-marketing" },
+    { platform: Platforms.Ios, size: { universal: 83.5 }, scale: 2, idiom: "ipad" },
+    { platform: Platforms.Ios, size: { universal: 20 }, scale: 1, idiom: "ipad" },
+    { platform: Platforms.Ios, size: { universal: 20 }, scale: 2, idiom: "ipad" },
+    { platform: Platforms.Ios, size: { universal: 29 }, scale: 1, idiom: "ipad" },
+    { platform: Platforms.Ios, size: { universal: 29 }, scale: 2, idiom: "ipad" },
+    { platform: Platforms.Ios, size: { universal: 40 }, scale: 1, idiom: "ipad" },
+    { platform: Platforms.Ios, size: { universal: 40 }, scale: 2, idiom: "ipad" },
+    { platform: Platforms.Ios, size: { universal: 76 }, scale: 1, idiom: "ipad" },
+    { platform: Platforms.Ios, size: { universal: 76 }, scale: 2, idiom: "ipad" },
+    { platform: Platforms.Ios, size: { universal: 20 }, scale: 2, idiom: "iphone" },
+    { platform: Platforms.Ios, size: { universal: 20 }, scale: 3, idiom: "iphone" },
+    { platform: Platforms.Ios, size: { universal: 29 }, scale: 2, idiom: "iphone" },
+    { platform: Platforms.Ios, size: { universal: 29 }, scale: 3, idiom: "iphone" },
+    { platform: Platforms.Ios, size: { universal: 40 }, scale: 2, idiom: "iphone" },
+    { platform: Platforms.Ios, size: { universal: 40 }, scale: 3, idiom: "iphone" },
+    { platform: Platforms.Ios, size: { universal: 60 }, scale: 2, idiom: "iphone" },
+    { platform: Platforms.Ios, size: { universal: 60 }, scale: 3, idiom: "iphone" },
+    { platform: Platforms.Android, size: { legacy: 192, adaptive: 432 }, dpi: "xxxhdpi" },
+    { platform: Platforms.Android, size: { legacy: 144, adaptive: 324 }, dpi: "xxhdpi" },
+    { platform: Platforms.Android, size: { legacy: 96, adaptive: 216 }, dpi: "xhdpi" },
+    { platform: Platforms.Android, size: { legacy: 72, adaptive: 162 }, dpi: "hdpi" },
+    { platform: Platforms.Android, size: { legacy: 48, adaptive: 108 }, dpi: "mdpi" },
+    { platform: Platforms.Expo, size: { mobile: 1024, web: 48 } },
+  ]
+}
+
+/**
+ * Validates that all necessary app-icon input files exist in the template dir.
+ * Additionally validates that they are of the correct size.
+ */
```

Comment:
Here we validate that:
- the provided CLI args are valid
- the input-files exist (for the selected option)
- the input-files are of the correct format (via file name)
- the input-files are of the correct size